static/app/utils/replays/useReplayData.tsx

```diff
@@ -109,55 +114,40 @@ function useReplayData({eventSlug, orgId}: Options): Result {
         const response = await api.requestPromise(
           `/api/0/projects/${orgId}/${projectId}/events/${eventId}/attachments/${attachment.id}/?download`
         );
-        return JSON.parse(response).events as eventWithTime;
+        return JSON.parse(response) as ReplayAttachment;
       })
     );
-    return attachments.flat();
-  }, [api, eventId, orgId, projectId]);
-
-  const fetchReplayEvents = useCallback(async () => {
-    const replayEventList = await api.requestPromise(
-      `/organizations/${orgId}/eventsv2/`,
-      {
-        query: {
-          statsPeriod: '14d',
-          project: [],
-          environment: [],
-          field: ['timestamp', 'replayId'],
-          sort: 'timestamp',
-          per_page: 50,
-          query: ['transaction:sentry-replay-event', `replayId:${eventId}`].join(' '),
-        },
-      }
-    );
 
-    return Promise.all(
-      replayEventList.data.map(
-        event =>
-          api.requestPromise(
-            `/organizations/${orgId}/events/${generateEventSlug(event)}/`
-          ) as Promise<Event>
-      )
-    );
-  }, [api, eventId, orgId]);
+    // ReplayAttachment[] => ReplayAttachment (merge each key of ReplayAttachment)
+    const result = attachments.reduce((acc, attachment) => {
+      Object.entries(attachment).forEach(([key, value]) => {
+        if (!acc[key]) {
+          acc[key] = [];
+        }
+        acc[key] = acc[key].concat(value);
+      });
+      return acc;
+    }, {}) as ReplayAttachment;
+    return result;
+  }, [api, eventId, orgId, projectId]);
 
   const loadEvents = useCallback(
     async function () {
       setState(INITIAL_STATE);
 
       try {
-        const [event, rrwebEvents, replayEvents] = await Promise.all([
+        const [event, attachments] = await Promise.all([
           fetchEvent(),
           fetchRRWebEvents(),
```

Comment:
I'm glad we haven't gotten into the incremental loading stuff yet.

It looks like what we can do now is make multiple `setState` calls when ajax data returns, but we would need to indicate in the UI `more is coming` vs. `we've got all the rrwebevents`